gsmphys/sfc_drv.f

```diff
@@ -566,6 +567,11 @@ subroutine sfc_drv                                                &
           tem     = 1.0 / rho(i)
           hflx(i) = hflx(i) * tem * cpinv
           evap(i) = evap(i) * tem * hvapi
+          if (evap(i) .lt. -maxevap(i)) then
+            chh(i) = -maxevap(i)/evap(i)*chh(i)
+            hflx(i) = -maxevap(i)/evap(i)*hflx(i)
```

Comment:
Checking order of operations is supposed to be `(maxevap/evap) * _var_` and you want to negate it.

Reply:
This looks mathematically equivalent to me. Compiler optimization might treat them differently.